drivers/modem/gsm_ppp.c

```diff
@@ -258,24 +260,44 @@ static struct setup_cmd setup_cmds2[] = {
 
 };
 
-MODEM_CMD_DEFINE(on_cmd_atcmdinfo_attached)
+MODEM_CMD_DEFINE(on_cmd_atcmdinfo_gprs_network_registration)
 {
-	int error = -EAGAIN;
-
-	/* Expected response is "+CGATT: 0|1" so simply look for '1' */
-	if (argc && atoi(argv[0]) == 1) {
-		error = 0;
-		LOG_INF("Attached to packet service!");
+	/* Expected response is "+CGREG: urc_enabled,status"
+	 * Status values:
+	 * 1 - registered, home network
+	 * 5 - registered, roaming
+	 */
+	if (argc > 1 && (atoi(argv[1]) == 1 || atoi(argv[1]) == 5)) {
+		gsm.gprs_registered = true;
+		LOG_INF("GPRS network registered.");
```

Comment:
Mise well print argv[1] too because it's sorta nice debug info

Reply:
done